grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -149,6 +149,46 @@ class UserService {
         return false
     }
 
+    Boolean hasRoleAssistantGlobally(User u) {
```

Comment:
Make all these methods generic. Add it to the User domain as a method:
```
Boolean hasDefaultRole(RoleType roleType) {
    return roles?.any { Role role -> role.roleType == roleType} ?: false
}
```
and then in the `LocationService` and in the `LocationApiController` you can do:
```
def inRoleBrowser = user.hasDefaultRole(RoleType.ROLE_BROWSER)
def inRoleAssistant = user.hasDefaultRole(RoleType.ROLE_ASSISTANT)
def inRoleManager = user.hasDefaultRole(RoleType.ROLE_MANAGER)
def inRoleAdmin = user.hasDefaultRole(RoleType.ROLE_ADMIN)
def inRoleSuperuser = user.hasDefaultRole(RoleType.ROLE_SUPERUSER)
```